iam/src/scripts/checkOnChainProvidersAreInSync.ts

```diff
@@ -0,0 +1,74 @@
+import dotenv from "dotenv";
+import { providers, Contract, version } from "ethers";
+import decoderAbi from "../../../deployments/abi/GitcoinPassportDecoder.json";
+
+import providerBitMapInfo from "../static/providerBitMapInfo.json";
+dotenv.config();
+
+console.log(process.argv);
+
+const apiUrl = process.argv[2] + process.env.ALCHEMY_API_KEY;
+const chainId = process.argv[3] as keyof typeof decoderAbi;
+const decoderContractAddress = process.argv[4];
+
+console.log("ethers version             :", version);
+console.log("apiUrl                     :", apiUrl);
+console.log("chainId                    :", chainId);
+console.log("decoderContractAddress     :", decoderContractAddress);
+
+function difference<T>(setA: Set<T>, setB: Set<T>): Set<T> {
+  const diff = new Set<T>(setA);
+  for (const elem of setB) {
+    diff.delete(elem);
+  }
+  return diff;
+}
+
+interface CustomContract extends Contract {
+  getProviders: (arg0: number) => Promise<string[]>;
+}
+
+async function main() {
+  let exitCode = 0;
+  const provider = new providers.JsonRpcProvider(apiUrl);
+  const decoderContract = new Contract(
+    decoderContractAddress,
+    decoderAbi[chainId],
+    provider
+  ) as unknown as CustomContract;
+
+  const decoderProviders = (await decoderContract.getProviders(0)) as unknown as string[];
```

Comment:
The version is hardcoded to 0 in the line above.
Ideally the current version should be pulled form the SC.